### PR TITLE
add psiflow

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jan-janssen** [comment]
While the code does not work on windows, I would still like the package to remain `noarch`:
```
  File "C:\bld\psiflow_1696617592318\_test_env\Lib\site-packages\parsl\multiprocessing.py", line 15, in <module>
    ForkProcess: Type = multiprocessing.get_context('fork').Process
```

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/psiflow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+source:
+  url: https://github.com/molmod/psiflow/archive/v{{ version }}.tar.gz
+  sha256: c6103905ecfe9ded67a1c52142424f184e50c24598998be931f25d5a13b6ae43
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - pmg = pymatgen.cli.pmg:main
```
Comment (synapticarbors) [inline recipes/psiflow/meta.yaml:17]: It seems unusual to have an entry point that comes in through a dependency. What's the reason for doing this?

File: recipes/psiflow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - wandb >=0.14.0
+    - pymatgen >=2023.3.23
+    - parsl ==2023.08.28
+
+test:
+  imports:
+    - psiflow
+  requires:
+    - pip
+  commands:
```
Comment (synapticarbors) [inline recipes/psiflow/meta.yaml:44]: Is it possible to add `commands` tests for all of the entry points?

Comment (jan-janssen) [reply recipes/psiflow/meta.yaml:52]: I tested all six commands but was only able to get three of them to work.

File: recipes/psiflow/meta.yaml (hunk 1 above)
Comment (jan-janssen) [reply recipes/psiflow/meta.yaml]: Sorry this was a mistake from my copy & pasting.

**Comment by jan-janssen** [comment]
@synapticarbors Can you take another look?

**Pull request merged**